doc/source/ray-air/examples/batch_tuning.ipynb

```diff
@@ -0,0 +1,1197 @@
+    "            (\"fare_amount\", \">\", 0),\n",
+    "            (\"pickup_location_id\", \"in\", [264, 265]),\n",
+    "            (\"dropoff_location_id\", \"not in\", [264, 265]), \n",
+    "            (\"dropoff_location_id\", \"=\", sample_id)\n",
+    "        ],\n",
+    "        columns=[\n",
+    "            \"pickup_at\",\n",
+    "            \"dropoff_at\",\n",
+    "            \"pickup_location_id\",\n",
+    "            \"dropoff_location_id\",\n",
+    "            \"passenger_count\",\n",
+    "            \"trip_distance\",\n",
+    "            \"fare_amount\",\n",
+    "        ],\n",
+    "    ).to_pandas()\n",
+    "\n",
+    "    return df\n",
+    "\n",
+    "# Function to transform a pandas dataframe\n",
+    "def transform_df(the_df: pd.DataFrame) -> pd.DataFrame:\n",
+    "    df = the_df.copy()\n",
+    "    \n",
+    "    df[\"trip_duration\"] = (df[\"dropoff_at\"] - df[\"pickup_at\"]).dt.seconds\n",
+    "    df = df[df[\"trip_duration\"] > 60]\n",
+    "    df = df[df[\"trip_duration\"] < 24 * 60 * 60] \n",
+    "    df.drop([\"dropoff_at\", \"pickup_at\", \"pickup_location_id\", \"fare_amount\"]\n",
+    "            , axis=1, inplace=True)\n",
+    "    df[\"dropoff_location_id\"] = df[\"dropoff_location_id\"].fillna(-1)\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72369cbf-e346-441c-b7e3-9280ccabc7f6",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Define your Ray Tune Search Space and Search Algorithm <a class=\"anchor\" id=\"define_search_space\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e310b13e-ba83-4c47-84e6-d1e916483b2b",
+   "metadata": {},
+   "source": [
+    "**First, define a search space of experiment to run.** \n",
+    "> The search space together with a search algorithm determine which experiments will be run.  \n",
+    "\n",
+    "Common search algorithms include grid search, random search, Bayesian search, Hyperopt, and Optuna.  For more details, see [Working with Tune Search Spaces](https://docs.ray.io/en/master/tune/tutorials/tune-search-spaces.html#tune-search-space-tutorial).  Deciding the best combination of search space and search algorithm is part of the art of being a Data Scientist and depends on the data, algorithm, and problem being solved!  \n",
+    "\n",
+    "**Below, we define our search space is:**\n",
+    "- 2 different Scikit-learn algorithms \n",
+    "- Some or all NYC taxi drop-off locations. \n",
+    "\n",
+    "**And we define the search algorithm is:**\n",
+    "- Grid search.\n",
+    "\n",
+    "> This means every permutation of each algorithm and each NYC Taxi drop-off location will be a separate experiment!  \n",
+    "\n",
+    "Ray Tune partitions the Search space using the specified Search algorithm and takes care of running a Tune job on each partition in parallel.  Specifically, Ray Tune will pass a config dictionary to each partition and make a Trainable function call."
```

Comment:
maybe something like
"Ray Tune will use the search space and the specified search algorithm to generate multiple configurations, each of which will be evaluated in a separate Trial on a Ray Cluster. Ray Tune will take care of orchestrating those Trials automatically."